controllers/admin/AdminCartRulesController.php

```diff
@@ -266,8 +266,10 @@ public function processAdd()
         if ($cart_rule = parent::processAdd()) {
             $this->context->smarty->assign('new_cart_rule', $cart_rule);
         }
+
         if (Tools::getValue('submitFormAjax')) {
             $this->redirect_after = false;
+            $this->context->smarty->assign('refresh_cart', true);
```

Comment:
For validation issue this could help (modal shouldn't close if validation fails).
if ($cart_rule) {
    $this->context->smarty->assign('refresh_cart', true);
}